tools/serverpod_cli/lib/src/analyzer/dart/endpoints_analyzer.dart

```diff
@@ -157,7 +161,7 @@ class EndpointsAnalyzer {
 
     var validationErrors = <String, List<SourceSpanSeverityException>>{};
     for (var classElement in endpointClasses) {
-      var errors = EndpointClassAnalyzer.validate(classElement);
+      var errors = EndpointClassAnalyzer.validate(classElement, []);
```

Comment:
Suggestion: Only do the class validation at this point.

This is the only place we should do the class validation. And we should only validate that we don't have a name collision with already validated classes. This ensures that the last defined endpoint will always have an error while the first defined class won't get this error.

So like you above have defined a list of `endpointDefs`,  you can add `classElements` to a list containing only the `classElement`s validated. Then the list is passed in as a parameter to the validate method that then validates the name is unique.